libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java

```diff
@@ -29,11 +29,12 @@
 /**
  * A connection.
  */
-public class FIXConnection<CHANNEL extends ReadableByteChannel & GatheringByteChannel> implements Closeable {
+public class FIXConnection implements Closeable {
 
     private static final int CURRENT_TIMESTAMP_FIELD_CAPACITY = 24;
 
-    private final CHANNEL channel;
+    private final ReadableByteChannel readableChannel;
+    private final GatheringByteChannel writableChannel;
```

Comment:
We are segregating the upstream and downstream channels - this serves two purposes - work around Java not having union types, and simplify cases like pipe transport, where channels are unidirectional.

Reply:
done